cli/src/cmd_net.rs

```diff
@@ -0,0 +1,1485 @@
+                enabled: false,
+                lldp_config: None,
+            },
+        };
+        match settings.links.get(PHY0) {
+            Some(_) => {
+                return Err(anyhow::anyhow!("only one link per port supported"));
+            }
+            None => {
+                settings.links.insert(String::from(PHY0), link);
+            }
+        }
+        ctx.client()?
+            .networking_switch_port_settings_create()
+            .body(settings)
+            .send()
+            .await?;
+        Ok(())
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "del")]
+pub struct CmdLinkDel {
+    /// Id of the rack to remove the link from.
+    rack: Uuid,
+
+    /// Switch to remove the link from.
+    #[arg(value_enum)]
+    switch: Switch,
+
+    /// Port to remove the link from.
+    #[arg(value_enum)]
+    port: Port,
+}
+
+#[async_trait]
+impl RunnableCmd for CmdLinkDel {
+    async fn run(&self, ctx: &Context) -> Result<()> {
+        let mut settings = current_port_settings(ctx, &self.rack, &self.switch, &self.port).await?;
+        settings.links.clear();
+        ctx.client()?
+            .networking_switch_port_settings_create()
+            .body(settings)
+            .send()
+            .await?;
+        Ok(())
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "bgp")]
+pub struct CmdBgp {
+    #[clap(subcommand)]
+    subcmd: BgpSubCommand,
+}
+
+#[async_trait]
+impl RunnableCmd for CmdBgp {
+    async fn run(&self, ctx: &Context) -> Result<()> {
+        match &self.subcmd {
+            BgpSubCommand::Status(cmd) => cmd.run(ctx).await,
+            BgpSubCommand::Config(cmd) => cmd.run(ctx).await,
+            BgpSubCommand::Announce(cmd) => cmd.run(ctx).await,
+            BgpSubCommand::Withdraw(cmd) => cmd.run(ctx).await,
+            BgpSubCommand::Filter(cmd) => cmd.run(ctx).await,
+        }
+    }
+}
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Parser, Debug, Clone)]
+enum BgpSubCommand {
+    /// Observe BGP status.
+    Status(CmdBgpStatus),
+
+    /// Manage BGP configuration.
+    Config(CmdBgpConfig),
+
+    /// Make a BGP announcement.
+    Announce(CmdBgpAnnounce),
+
+    /// Make a BGP announcement.
+    Withdraw(CmdBgpWithdraw),
+
+    /// Set a filtering specification for a peer.
+    Filter(CmdBgpFilter),
+}
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "announce")]
+pub struct CmdBgpAnnounce {
+    /// The announce set to announce from.
+    announce_set: Name,
+
+    /// The address lot to draw from.
+    address_lot: Name,
+
+    /// The prefix to announce.
+    prefix: oxnet::IpNet,
+}
+
+#[async_trait]
+impl RunnableCmd for CmdBgpAnnounce {
+    async fn run(&self, ctx: &Context) -> Result<()> {
+        let mut current: Vec<BgpAnnouncementCreate> = ctx
+            .client()?
+            .networking_bgp_announce_set_list()
```

Comment:
@internet-diglett since you're looking into the networking API, these "list" APIs should probably use paginated results